feat(web): implement no-fee november front end designs

Zalens opened the pull request:
## What it solves

Solves https://linear.app/safe-global/issue/REVE-540/featweb-add-front-end-designs-for-no-fee-november and all sub-issues

## How this PR fixes it

- **No Fee November Banner** on Home, Assets page and Creation Tx flow: Created banner component
- **Eligibility Hook**: `useNoFeeNovemberEligibility` with CGW endpoint
- **State Management**: Enhanced dismissal state to track `{ dismissed: boolean, lastEligibilityState: boolean, dismissedAt: number }`
- **No Fee November Execution Selector**:  Needed to be updated to support the No-fee November.

## How to test it

1. Navigate to any Safe on Mainnet (chainId: '1')
2. Banner appears with loading skeleton initially
3. It shows either the banner if user eligible either nothing
4. Click the close button to dismiss the banner
5. Banner reappears if eligibility state changes after dismissal
6. Navigate to Assets
7. Initiate a transaction to view the banner
8. Execute a transaction to see the execution selector method

To test it currently, you must use the PR branch: https://github.com/safe-global/safe-client-gateway/pull/2753 of safe-client-gateway to get the correct behavoir. 

## Screenshots
<img width="1656" height="1173" alt="Screenshot 2025-10-21 at 16 39 17" src="https://github.com/user-attachments/assets/2ef717df-7f89-4fcf-94ac-d92d1897b527" />
<img width="1602" height="1011" alt="Screenshot 2025-10-23 at 12 10 24" src="https://github.com/user-attachments/assets/c79e078b-d90e-4372-9014-34931ceb231a" />
<img width="1636" height="1184" alt="Screenshot 2025-10-28 at 09 46 40" src="https://github.com/user-attachments/assets/32664c8d-2d34-4be8-94e2-a0e57ed34c81" />
<img width="1636" height="1187" alt="Screenshot 2025-10-28 at 09 55 02" src="https://github.com/user-attachments/assets/af3c6e47-82a3-4887-a6ec-d42ed26d128b" />
<img width="1637" height="1183" alt="Screenshot 2025-10-28 at 09 55 28" src="https://github.com/user-attachments/assets/f56561b3-ee4c-4c78-ab22-2cca168671a5" />
<img width="1631" height="1185" alt="Screenshot 2025-10-28 at 09 58 27" src="https://github.com/user-attachments/assets/2e3ba3ea-100a-4126-8c28-fd90fd8dc4b9" />


## Checklist

- [] I've tested the branch on mobile 📱
- [] I've documented how it affects the analytics (if at all) 📊
- [] I've written a unit/e2e test for it (if applicable) 🧑‍💻

## Notes
- Used the relayer behavior to maintain cleaner and more organized code for no-fee November - The current design is not fully aligned with Figma due to those choices.
- Backend PR: https://github.com/safe-global/safe-client-gateway/pull/2753

---

## CLA signature

With the submission of this Pull Request, I confirm that I have read and agree to the terms of the [Contributor License Agreement](https://safe.global/cla).

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Implements No‑Fee November UI and logic: eligibility-driven banners, sponsored gas execution option, enhanced banner dismissal, and gas/limit handling across dashboard, assets, and tx flows.
> 
> - **No‑Fee November**:
>   - Add dashboard and Assets page banner (`NoFeeNovemberBanner`) and tx flow card (`NoFeeNovemberTransactionCard`) with learn‑more links and CTA.
>   - Integrate feature flag `FEATURES.NO_FEE_NOVEMBER` and eligibility hook `useNoFeeNovemberEligibility` (uses relays remaining endpoint, blocked address handling, refetching).
>   - New gas limit rule (`MAX_GAS_LIMIT_NO_FEE_CAMPAIGN`) and `useGasTooHigh` with info banner (`GasTooHighBanner`).
> - **Execution Flow**:
>   - Extend `ExecutionMethodSelector` with `NO_FEE_NOVEMBER` option, availability/limit display, disabled states when gas too high or limit reached, and remaining counter.
>   - Show "Free / No‑Fee November" in `GasParams`; wire through via `AdvancedParams` and `ExecuteForm` (decides relaying vs no‑fee path and submit logic).
> - **Banner Dismissal & Carousel**:
>   - Enhance dismissal state to support `{ dismissed, lastEligibilityState, dismissedAt }` and re‑show on eligibility change (`isBannerDismissed`, `dismissBanner`).
>   - Update `NewsCarousel`/`NewsDisclaimers` to use new dismissal shape and `eligibilityState` per banner.
> - **Balances Page**:
>   - Show sticky `NoFeeNovemberBanner` with localStorage dismissal.
> - **Assets/Media**:
>   - Add SVG assets under `public/images/common/no-fee-november/`.
> - **Utilities**:
>   - Add `useIsNoFeeNovemberFeatureEnabled`; update `useRemainingRelays` to refetch more aggressively; extend `FEATURES` enum with `NO_FEE_NOVEMBER`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 73f157e0262d8a195d3cc6de06fca8ca94d57ab0. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

Zalens committed:
feat: add NO_FEE_NOVEMBER feature flag

Zalens committed:
feat: add useNoFeeNovemberEligibility hook with mock implementation

Zalens committed:
feat: add useIsNoFeeNovemberBannerVisible hook

Zalens committed:
feat: add SafeCoinsIllustration component

Zalens committed:
feat: add No Fee November banner assets

Zalens committed:
feat: add NoFeeNovemberBanner component with eligible/not eligible states

Zalens committed:
feat: add enhanced dismissal state tracking for banner eligibility

Zalens committed:
feat: update NewsCarousel to support eligibility state tracking

Zalens committed:
feat: update NewsDisclaimers to use enhanced dismissal logic

Zalens committed:
feat: integrate No Fee November banner into dashboard

Zalens committed:
test: add comprehensive tests for No Fee November banner functionality

Zalens committed:
feat: implement No Fee November banner on Assets page (#6517)

* feat: add No Fee November banner to Assets page with sticky positioning

* fix lint errors on index.tsx

* refactor: encapsulate No Fee November feature

katspaugh commented on apps/web/public/images/common/no-fee-november/safe-token.png:
There are three identical files with different names, is that expected?

katspaugh replied:
Why do you need 3 identical images? Just use the same one.

Also I see 3 identical backgrounds. Gradients don't even need to be an image, it can be done in CSS.

valens-carpentier replied:
@IvaLukan provided an SVG file. I'll make the adjustments.

katspaugh commented on apps/web/src/components/common/SafeCoinsIllustration/index.tsx:
```diff
@@ -0,0 +1,245 @@
+        height: '76px',
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
```
Can you move all the inline styles in the file into a CSS module?

katspaugh commented on apps/web/src/features/no-fee-november/hooks/useIsNoFeeNovemberEnabled.ts:
```diff
@@ -0,0 +1,25 @@
+ * TODO: Remove mock when feature flag is enabled on Mainnet chain configuration
+ */
+const useIsNoFeeNovemberEnabled = (): boolean => {
+  // const isFeatureEnabled = useHasFeature(FEATURES.NO_FEE_NOVEMBER)
```
I've enabled it on all chains on staging for you.

Zalens committed:
feat: add No Fee November card to token transfer create step (#6519)

Zalens committed:
feat(web): add no fee november select or in execution method (#6523)

* fix: improve No-fee November UI styling

- Match transaction counter styling with wallet balance
- Add proper spacing between number and text
- Use design system tokens for colors and borders

* feat(web): add no fee november selector in execution method

* lint(web): fix lint errors

Zalens committed:
refactor: move SafeCoinsIllustration inline styles to CSS module

Zalens committed:
fix: adjust Safe logo positioning and banner text spacing

Zalens committed:
fix: update test mocks to include missing hook properties

Zalens committed:
replace coin illustrations with proper svg and disable buttons when wallet not connected

Zalens committed:
feat: delete No Fee November banner for non-eligible users

Zalens committed:
adapt transaction banner new style

Zalens committed:
update design for banner in home

Zalens committed:
modify transaction banner creation flow card

Zalens committed:
update execute no fee november page

Zalens committed:
feat(web): implement geofencing and blacklist checks for No Fee November feature

Zalens committed:
feat(web): implement gas too high views for No Fee November with disabled state and info banner

Zalens committed:
feat(web): update no-fee november with working backend endpoints

Zalens committed:
Merge branch 'dev' into feat/no-fee-november-banner

Zalens committed:
remove duplicate gradient on banner and solve type check error

Zalens committed:
remove legacy images for no-fee banner

Zalens committed:
remove former illustration for banner

Zalens committed:
add hyperlink to terms

Zalens committed:
improve dark mode UI and remove TCs logic

Zalens committed:
remove NO_FEE_NOVEMBER from FEATURES

Zalens committed:
chores: change wording for transactions left

valens-carpentier commented on apps/web/src/features/no-fee-november/components/NoFeeNovemberBanner/index.tsx:
```diff
@@ -0,0 +1,72 @@
+
+export const noFeeNovemberBannerID = 'noFeeNovemberBanner'
+
+const NoFeeNovemberBanner = ({ onDismiss }: { onDismiss: () => void }) => {
```
Do you think this is necessary now? QA is working on the branch. 

If we run the campaign again in the future, we should do it, of course. 

But there is no certainty that we are going to run it again. Anyway, those changes will not be heavy in the future if we need to do so.

Zalens committed:
fix eligibility logic on hook

Zalens committed:
fix: solve QA remarks

Zalens committed:
fix: add refetch on transaction limit

Zalens committed:
feat: add no fee november feature flag

Zalens committed:
fix feature flag for execution method selector

Zalens committed:
fix: add check on chain for eligibility hook
